features/sharding/plugin/cosid/src/main/java/org/apache/shardingsphere/sharding/cosid/algorithm/keygen/CosIdSnowflakeKeyGenerateAlgorithm.java

```diff
@@ -73,7 +73,10 @@ private boolean getAsString(final Properties props) {
     }
     
     private long getEpoch(final Properties props) {
-        return Long.parseLong(props.getProperty(EPOCH_KEY, DEFAULT_EPOCH + ""));
+        long result = Long.parseLong(props.getProperty(EPOCH_KEY, DEFAULT_EPOCH + ""));
+        ShardingSpherePreconditions.checkState(result > 0L,
+                () -> new ShardingPluginException("Key generate algorithm `%s` initialization failed, reason is: %s.", getType(), "Epoch must be positive."));
```

Comment:
Should use KeyGenerateAlgorithmInitializationException here.

Reply:
Thanks! If I use org.apache.shardingsphere.sharding.exception.algorithm.keygen.KeyGenerateAlgorithmInitializationException here, I need to add dependency on module  "shardingsphere-sharding-core", which will cause a circular dependency between "shardingsphere-sharding-cosid" and "shardingsphere-sharding-core". May I create a new KeyGenerateAlgorithmInitializationException class in the same package as ShardingPluginException to solve this problem?

Reply:
> Should use KeyGenerateAlgorithmInitializationException here.

Thanks! If I use org.apache.shardingsphere.sharding.exception.algorithm.keygen.KeyGenerateAlgorithmInitializationException here, I need to add dependency on module "shardingsphere-sharding-core", which will cause a circular dependency between "shardingsphere-sharding-cosid" and "shardingsphere-sharding-core". May I create a new KeyGenerateAlgorithmInitializationException class in the same package as ShardingPluginException to solve this problem?

Reply:
> Should use KeyGenerateAlgorithmInitializationException here.

@Pace2Car Thanks! If I use org.apache.shardingsphere.sharding.exception.algorithm.keygen.KeyGenerateAlgorithmInitializationException here, I need to add dependency on module "shardingsphere-sharding-core", which will cause a circular dependency between "shardingsphere-sharding-cosid" and "shardingsphere-sharding-core". May I create a new KeyGenerateAlgorithmInitializationException class in the same package as ShardingPluginException to solve this problem?

Reply:
@shengdoupi Sorry, I forgot to check the dependencies, so it's fine to keep it as it is.